flang/lib/Optimizer/Builder/IntrinsicCall.cpp

```diff
@@ -147,6 +147,10 @@ static constexpr IntrinsicHandler handlers[]{
     {"atan2pi", &I::genAtanpi},
     {"atand", &I::genAtand},
     {"atanpi", &I::genAtanpi},
+    {"atomicaddd", &I::genAtomAdd, {{{"addr", asAddr}, {"v", asValue}}}, false},
```

Comment:
At some point, we should maybe move the CUF intrinsic module procedures into there own table (a bit like what is done for ppc intrinsics).

This  would clarify what are regular Fortran intrinsics expected to be intercepted here, and what are CUF extensions (and if well done, could speed-up the lookup by only looking into the right table based on the intrinsic module).

You can leave that alone, I will try to split the table later (maybe ieee proc could also be taken out).